frontend_integration_library.md

```diff
@@ -0,0 +1,154 @@
+# Metrilo's JS tracking library
+
+:warning: **We strongly recommend doing this on Staging/Beta environment first to make sure the integration is smooth and nothing breaks down.**
+
+Metrilo provides a javascript library for integration with your custom e-commerce solution. It helps you track the events that the people visiting your shop do, such as viewing specific pages or products, adding products to the cart, making orders, etc. The provided data is transformed into different reports and analytics metrics you could benefit from.
+For more information about Metrilo and its features please visit [the official website](https://www.metrilo.com/).
+
+## Integration
+### Installing the JS library
+Insert the following script tag in the `<head>` tag of the page you want to track.
+```
+<script type="text/javascript" src="https://tracking.metrilo.com/tracking.js?token=<YOUR_PROJECT_TOKEN>"></script>
+```
+Consider the `<YOUR_PROJECT_TOKEN>` parameter - here you should put your project's API token, which you could find on the Settings page of your project in Metrilo.
+
+:warning: **You should add the script tag on every page you want to track!**
+
+## Tracking
+In order to start tracking visitor activity you have to call the functions defined in the `window.metrilo` object. See the sections below for more information.
+* [Identify a user](#identify-a-user)
+* [Tracking events](#tracking-events)
+
+### Tracking responses
+The following responses are valid for all of the events that the tracking library provides.
+
+| Code | Message |  
+| --- | --- |  
+| 204 | Success response. |  
+| 400 | Bad request from client. |  
+| 401 | Project token is invalid. |  
+| 402 | Project requires payment. |  
+| 403 | The IP that the request came from is ignored. |  
+| 500 | Error from server while processing request. |  
+| 502 | Error from server accepting request. |  
+
+### Identify a user
+When a customer comes to the store and he is not logged in Metrilo treats him as an anonymous visitor. Metrilo adds a unique identifier to his session, so it could attribute his actions to this identifier. Once he provides his email (e.g. logs in, signs up, fills in a subscription form, makes an order, etc.) you have to call ```metrilo.identify(customerEmail)```. This will associate the unique identifier with the email. Moreover, all events that the visitor has while being an anonymous one will be merged with the events he has as identified (*if any). Keep in mind that the identify call must be done only once, the first time the visitor enters his email. If you are not sure that you have already called it, just check for the existence of email in the cookie named **cbuid**. If there is an email, no need to call it. This cookie contains either the unique identifier or the email and part of the identify function is to change the content  from the unique identifier to the email.
```

Comment:
Can you remind me why is it bad to call identify more than once?

Reply:
once the identified is made, the cbuid cookie is changed with the email. If you try to call metrilo.identify(email) again, it will throw an error "missing visitorId", because there is no visitorId anymore and you could not possible do identify

Reply:
Correct me if I'm wrong, this error will only be seen in the 4XX response to the request made by the JS tracking library?

Reply:
yeah, it will be only in the browser

Reply:
Maybe we should handle it on the JS library level: do not send the identify if the email is already in the cookie?
https://github.com/Metrilo/tracking/blob/master/src/main/js/tracking.js#L90